SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -204,24 +215,38 @@ bool ValidateIdentifierNames(TArray<TSharedPtr<FUnrealType>>& TypeInfos)
 		UClass* Class = Cast<UClass>(TypeInfo->Type);
 		check(Class);
 		const FString& ClassName = Class->GetName();
+		const FString& ClassPath = Class->GetPathName();
 		FString SchemaName = UnrealNameToSchemaName(ClassName);
 
-		if (!CheckSchemaNameValidity(SchemaName, Class->GetPathName(), TEXT("Class")))
+		if (!CheckSchemaNameValidity(SchemaName, ClassPath, TEXT("Class")))
 		{
 			bSuccess = false;
 		}
 
+		AddNameCollision(SchemaName, ClassPath);
+		
+		if (ClassToSchemaName.Contains(ClassPath))
+		{
+			continue;
+		}
+
 		int Suffix = 0;
 		while (UsedSchemaNames.Contains(SchemaName))
 		{
-			UE_LOG(LogSpatialGDKSchemaGenerator, Warning, TEXT("Class name collision after removing non-alphanumeric characters. Name '%s' collides for types '%s' and '%s'. Will add a suffix to resolve the collision."),
-				*SchemaName, *Class->GetPathName(), *UsedSchemaNames[SchemaName]->GetPathName());
-
 			SchemaName = UnrealNameToSchemaName(ClassName) + FString::Printf(TEXT("%d"), ++Suffix);
 		}
 
-		ClassToSchemaName.Add(Class, SchemaName);
-		UsedSchemaNames.Add(SchemaName, Class);
+		ClassToSchemaName.Add(Class->GetPathName(), SchemaName);
+		UsedSchemaNames.Add(SchemaName, Class->GetPathName());
+	}
+
+	for (auto& Collision : NameCollisions)
+	{
+		if (Collision.Value.Num() > 1)
+		{
+			UE_LOG(LogSpatialGDKSchemaGenerator, Warning, TEXT("Class name collision after removing non-alphanumeric characters. Name '%s' collides for types [%s]"),
```

Comment:
Should we indicate that we resolved the collision by appending numbers and showing what maps to what now?

Reply:
I think we should.